services/spar/src/Spar/API.hs

```diff
@@ -228,15 +228,27 @@ idpGetAll zusr = withDebugLog "idpGetAll" (const Nothing) $ do
   _idplProviders <- wrapMonadClientWithEnv $ Data.getIdPConfigsByTeam teamid
   pure IdPList {..}
 
-idpDelete :: Maybe UserId -> SAML.IdPId -> Spar NoContent
-idpDelete zusr idpid = withDebugLog "idpDelete" (const Nothing) $ do
+idpDelete :: Maybe UserId -> SAML.IdPId -> Maybe Bool -> Spar NoContent
+idpDelete zusr idpid purge = withDebugLog "idpDelete" (const Nothing) $ do
```

Comment:
The problem is this will never work for large teams. it will always fail halfway through and timeout.  We need something better for this.

Reply:
a4fd2705beef216bfc0b412ad6be45ec10ea6a79